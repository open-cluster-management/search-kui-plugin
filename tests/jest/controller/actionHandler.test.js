/*******************************************************************************
* Licensed Materials - Property of IBM
* (c) Copyright IBM Corporation 2019. All Rights Reserved.
*
* Note to U.S. Government Users Restricted Rights:
* Use, duplication or disclosure restricted by GSA ADP Schedule
* Contract with IBM Corp.
*******************************************************************************/
'use strict'
const actionHandler = require('../../../dist/src-web/controller/actionHandler')

describe('Search action handler', () => {
  const name = 'All Fake Pods'
  const notification = {
    success: {
      content: `Successfully deleted saved search ${name}`
    },
    warning: {
      warning: `Warning - saved search ${name} was not found! Check saved searches`
    },
    error: {
      message: `Failed to delete saved search ${name}`,
    }
  }

  describe('Delete saved searches', () => {
    const args =  {
      command: `deleteSavedSearch ${name}`,
      argv: ['deleteSavedSearch', name]
    }

    const spy = jest.spyOn(actionHandler, 'deleteSavedSearch')
    const _ = actionHandler.deleteSavedSearch(args)

    it('should delete a saved search', () => {
      expect(spy).toBeCalled()
      expect(actionHandler.deleteSavedSearch).toMatchSnapshot()
    })
  })

  describe('Delete resources', () => {
    const args =  {
      command: 'deleteResource fake-name fake-namespace fake-kind fake-cluster fake-selfLink',
      argv: ['deleteResource', 'fake-name', 'fake-namespace', 'fake-kind', 'fake-cluster', 'fake-selfLink']
    }

    const spy = jest.spyOn(actionHandler, 'deleteResource')
    const _ = actionHandler.deleteResource(args)

    it('should delete a resource', () => {
      expect(spy).toBeCalled()
      expect(actionHandler.deleteResource).toMatchSnapshot()
    })
  })

  // describe('Notifications', () =>  {
  //   it('should return a success notification', () => {
  //     expect(actionHandler.notify(notification.success.content)).toMatchSnapshot()
  //   })

  //   it('should return a warning notification', () => {
  //     expect(actionHandler.notify(notification.warning)).toMatchSnapshot()
  //   })

  //   it('should return a error notification', () => {
  //     expect(actionHandler.notify(notification.error)).toMatchSnapshot()
  //   })
  // })
})
