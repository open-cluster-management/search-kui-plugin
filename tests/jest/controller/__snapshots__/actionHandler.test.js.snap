// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search action handler Delete resources should delete a resource 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "argv": Array [
          "deleteResource",
          "fake-name",
          "fake-namespace",
          "fake-kind",
          "fake-cluster",
          "fake-selfLink",
        ],
        "command": "deleteResource fake-name fake-namespace fake-kind fake-cluster fake-selfLink",
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": Promise {},
    },
  ],
}
`;

exports[`Search action handler Delete saved searches should delete a saved search 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "argv": Array [
          "deleteSavedSearch",
          "All Fake Pods",
        ],
        "command": "deleteSavedSearch All Fake Pods",
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": Promise {},
    },
  ],
}
`;

exports[`Search action handler Notifications should return a error notification 1`] = `
<div
  class="bx--toast-notification-content"
>
  <div
    class="notification"
  >
    <div
      class="bx--toast-notification bx--toast-notification--error"
      kind="error"
      role="alert"
    >
      <svg
        aria-hidden="true"
        class="bx--toast-notification__icon"
        focusable="false"
        height="20"
        preserveAspectRatio="xMidYMid meet"
        style="will-change: transform;"
        viewBox="0 0 20 20"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M10,1c-5,0-9,4-9,9s4,9,9,9s9-4,9-9S15,1,10,1z M13.5,14.5l-8-8l1-1l8,8L13.5,14.5z"
        />
        <path
          d="M13.5,14.5l-8-8l1-1l8,8L13.5,14.5z"
          data-icon-path="inner-path"
          opacity="0"
        />
        <title>
          error icon
        </title>
      </svg>
      <div
        class="bx--toast-notification__details"
      >
        <h3
          class="bx--toast-notification__title"
        >
          Failed to delete saved search All Fake Pods
        </h3>
        <div
          class="bx--toast-notification__subtitle"
        />
        <div
          class="bx--toast-notification__caption"
        >
          12:00:00 PM
        </div>
      </div>
      <button
        aria-label="closes notification"
        class="bx--toast-notification__close-button"
        title="closes notification"
        type="button"
      >
        <svg
          aria-label="close notification"
          class="bx--toast-notification__close-icon"
          focusable="false"
          height="20"
          preserveAspectRatio="xMidYMid meet"
          role="img"
          style="will-change: transform;"
          viewBox="0 0 32 32"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M24 9.4L22.6 8 16 14.6 9.4 8 8 9.4 14.6 16 8 22.6 9.4 24 16 17.4 22.6 24 24 22.6 17.4 16 24 9.4z"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;

exports[`Search action handler Notifications should return a success notification 1`] = `
<div
  class="bx--toast-notification-content"
>
  <div
    class="notification"
  >
    <div
      class="bx--toast-notification bx--toast-notification--success"
      kind="success"
      role="alert"
    >
      <svg
        aria-hidden="true"
        class="bx--toast-notification__icon"
        focusable="false"
        height="20"
        preserveAspectRatio="xMidYMid meet"
        style="will-change: transform;"
        viewBox="0 0 20 20"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M10,1c-4.9,0-9,4.1-9,9s4.1,9,9,9s9-4,9-9S15,1,10,1z M8.7,13.5l-3.2-3.2l1-1l2.2,2.2l4.8-4.8l1,1L8.7,13.5z"
        />
        <path
          d="M8.7,13.5l-3.2-3.2l1-1l2.2,2.2l4.8-4.8l1,1L8.7,13.5z"
          data-icon-path="inner-path"
          fill="none"
          opacity="0"
        />
        <title>
          success icon
        </title>
      </svg>
      <div
        class="bx--toast-notification__details"
      >
        <h3
          class="bx--toast-notification__title"
        >
          Successfully deleted saved search All Fake Pods
        </h3>
        <div
          class="bx--toast-notification__subtitle"
        />
        <div
          class="bx--toast-notification__caption"
        >
          12:00:00 PM
        </div>
      </div>
      <button
        aria-label="closes notification"
        class="bx--toast-notification__close-button"
        title="closes notification"
        type="button"
      >
        <svg
          aria-label="close notification"
          class="bx--toast-notification__close-icon"
          focusable="false"
          height="20"
          preserveAspectRatio="xMidYMid meet"
          role="img"
          style="will-change: transform;"
          viewBox="0 0 32 32"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M24 9.4L22.6 8 16 14.6 9.4 8 8 9.4 14.6 16 8 22.6 9.4 24 16 17.4 22.6 24 24 22.6 17.4 16 24 9.4z"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;

exports[`Search action handler Notifications should return a warning notification 1`] = `
<div
  class="bx--toast-notification-content"
>
  <div
    class="notification"
  >
    <div
      class="bx--toast-notification bx--toast-notification--warning"
      kind="warning"
      role="alert"
    >
      <svg
        aria-hidden="true"
        class="bx--toast-notification__icon"
        focusable="false"
        height="20"
        preserveAspectRatio="xMidYMid meet"
        style="will-change: transform;"
        viewBox="0 0 20 20"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M10,1c-5,0-9,4-9,9s4,9,9,9s9-4,9-9S15,1,10,1z M9.2,5h1.5v7H9.2V5z M10,16c-0.6,0-1-0.4-1-1s0.4-1,1-1	s1,0.4,1,1S10.6,16,10,16z"
        />
        <path
          d="M9.2,5h1.5v7H9.2V5z M10,16c-0.6,0-1-0.4-1-1s0.4-1,1-1s1,0.4,1,1S10.6,16,10,16z"
          data-icon-path="inner-path"
          opacity="0"
        />
        <title>
          warning icon
        </title>
      </svg>
      <div
        class="bx--toast-notification__details"
      >
        <h3
          class="bx--toast-notification__title"
        >
          Warning - saved search All Fake Pods was not found! Check saved searches
        </h3>
        <div
          class="bx--toast-notification__subtitle"
        />
        <div
          class="bx--toast-notification__caption"
        >
          12:00:00 PM
        </div>
      </div>
      <button
        aria-label="closes notification"
        class="bx--toast-notification__close-button"
        title="closes notification"
        type="button"
      >
        <svg
          aria-label="close notification"
          class="bx--toast-notification__close-icon"
          focusable="false"
          height="20"
          preserveAspectRatio="xMidYMid meet"
          role="img"
          style="will-change: transform;"
          viewBox="0 0 32 32"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M24 9.4L22.6 8 16 14.6 9.4 8 8 9.4 14.6 16 8 22.6 9.4 24 16 17.4 22.6 24 24 22.6 17.4 16 24 9.4z"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;
